File: csharp/main/Float64Def.cs
using System;
using System.Collections.Generic;

namespace Deltix.Luminary
{
	public class Float64Def : IConstantScope, ITypeDef<TypeFloat64>
	{
		private Float64Def()
		{
			DefinedConstants.Add(new ConstantDef(this, "MIN_VALUE", TypeFloat64.Instance, new LiteralFloat64(Double.MinValue)));
			DefinedConstants.Add(new ConstantDef(this, "MAX_VALUE", TypeFloat64.Instance, new LiteralFloat64(Double.MaxValue)));
			DefinedConstants.Add(new ConstantDef(this, "NaN", TypeFloat64.Instance, new LiteralFloat64(Double.NaN)));
			DefinedConstants.Add(new ConstantDef(this, "POSITIVE_INFINITY", TypeFloat64.Instance, new LiteralFloat64(Double.PositiveInfinity)));
			DefinedConstants.Add(new ConstantDef(this, "NEGATIVE_INFINITY", TypeFloat64.Instance, new LiteralFloat64(Double.NegativeInfinity)));
		}

		public static Float64Def Instance = new Float64Def();

		public String Name => "Float64";

		public String FullName => Name;

		public List<ConstantDef> DefinedConstants { get; } = new List<ConstantDef>();

		public TypeFloat64 Type => TypeFloat64.Instance;

		public FileDef File => null;

		public ITypeScope Parent => null;
	}
}

File: java/src/main/java/deltix/luminary/Float64Def.java
package deltix.luminary;

import java.util.ArrayList;
import java.util.List;

public class Float64Def implements TypeDef<TypeFloat64>, ConstantScope {
    private List<ConstantDef> definedConstants;

    private Float64Def() {
        definedConstants = new ArrayList<>();
        definedConstants.add(new ConstantDef(this, "MIN_VALUE", TypeFloat64.INSTANCE, new LiteralFloat64(Double.MIN_VALUE)));
        definedConstants.add(new ConstantDef(this, "MAX_VALUE", TypeFloat64.INSTANCE, new LiteralFloat64(Double.MAX_VALUE)));
        definedConstants.add(new ConstantDef(this, "NaN", TypeFloat64.INSTANCE, new LiteralFloat64(Double.NaN)));
        definedConstants.add(new ConstantDef(this, "POSITIVE_INFINITY", TypeFloat64.INSTANCE, new LiteralFloat64(Double.POSITIVE_INFINITY)));
        definedConstants.add(new ConstantDef(this, "NEGATIVE_INFINITY", TypeFloat64.INSTANCE, new LiteralFloat64(Double.NEGATIVE_INFINITY)));
    }

    public static final Float64Def INSTANCE = new Float64Def();

    @Override
    public String getName() {
        return "Float64";
    }

    @Override
    public String getFullName() {
        return getName();
    }

    @Override
    public List<ConstantDef> getDefinedConstants() {
        return definedConstants;
    }

    @Override
    public TypeFloat64 getType() {
        return TypeFloat64.INSTANCE;
    }

    @Override
    public FileDef getFile() {
        return null;
    }

    @Override
    public TypeScope getParent() {
        return null;
    }
}
